msticpy/context/azure/sentinel_core.py

```diff
@@ -35,6 +39,59 @@
 
 logger = logging.getLogger(__name__)
 
+_SUB_ID = "subscription_id"
+_RES_GRP = "resource_group"
+_WS_NAME = "workspace_name"
+_RES_ID = "resource_id"
+
+
+@dataclass
+class SentinelInstanceDetails:
+    """Dataclass for Sentinel workspace details."""
+
+    subscription_id: Optional[str]
```

Comment:
I am not a fan of making these optional. The point of this pattern is that the `SentinelInstanceDetails` can only instantiate when it has the necessary arguments. If any of these are not passed, throwing an exception is the right thing to do.